sdk/highlight-next/src/util/highlight-middleware.ts

```diff
@@ -0,0 +1,9 @@
+import type { NextRequest } from 'next/server'
+
+export function highlightMiddleware(request: NextRequest) {
+	const sessionSecureID = request.cookies.get('sessionSecureID')?.value
```

Comment:
Middleware checks for the existence of the cookie and a missing `x-highlight-request` header, and attempts to add the `x-highlight-request` header if appropriate.

There's no requestId here, so I'm unsure if it causes a problem. It'll be undefined in `H.runWithHeaders`